epix/run_epix.py

```diff
@@ -0,0 +1,202 @@
+import os
+import time
+import awkward as ak
+import numpy as np
+import pandas as pd
+import warnings
+
+import epix
+
+
+def main(args, return_df=False, return_wfsim_instructions=False, strax=False):
+    """Call this function from the run_epix script"""
+
+    if args['debug']:
+        print("epix configuration: ", args)
+        # TODO: also add memory information see starxer and change this to debug
+        # Getting time information:
+        starttime = time.time()
+        tnow = starttime
+
+    # Loading data:
+    inter, n_simulated_events = epix.loader(args['path'],
+                                            args['file_name'],
+                                            args['debug'],
+                                            outer_cylinder=args['outer_cylinder'],
+                                            kwargs_uproot_arrays={'entry_start': args['entry_start'],
+                                                                  'entry_stop': args['entry_stop']}
+                                            )
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'load data.')
+        print(f"Finding clusters of interactions with a dr = {args['micro_separation']} mm"
+               f" and dt = {args['micro_separation_time']} ns")
+
+    # Cluster finding and clustering:
+    inter = epix.find_cluster(inter, args['micro_separation']/10, args['micro_separation_time'])
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'cluster finding.')
```

Comment:
Just a minor comment: This will print something like "It took x seconds to cluster finding." Not an expert but this loos grammatically wrong to me ;-)

Reply:
Thumbs up!